### ISSUE TITLE
6/27 ~ 6/30 : 0주차 정리

### ISSUE THREAD

**Issue by jwkwak45** [jwkwak45/AIstudy.github.io#2]
6/28 [실습] 기본 파이썬 문법 실습

기본적인 파이썬 내용이라 그냥 앞으로는 주피터 노트북을 쓰게 될 것이라는 걸 확인하고 편하게 들어도 상관 없는 내용이었습니다. Numpy 부분만 조금 더 열심히 들었습니다.

**Comment by jwkwak45** [comment]
6/30 기계 학습 용어 정리

Output / Class / Label : 결과물
--> One-hot coding : 1 or 0

Dataset 
Training Data : 직접적인 학습에 필요
Validation Data : Hyper parameter 를 위해 필요
Test Data : 트레이닝과 무관해야 함

Basic single Layer network
Neural Network
Multi-Layer Perceptron
Activation Function : Non-Linear 성을 위함

Epoch : 전체 데이터를 다 확인하는것
Batch size : 한번 계산에 확인하는 데이터 수
Iteration : 전체 데이터를 계산할 때 필요한 횟수

Cost function : 줄이고자 하는 함수

MNIST : 이 강의에서 앞으로 학습할 때 사용할 데이터셋 (원래 뜻은 다름)

**Comment by YunjeongPARK1** [comment]
기계학습 용어 정리 보충

*Neural Network 이란
 : 기본적으로 연결이 되어 있는 구조
 : n space에서 m space로 가는 nonlinear mapping을 찾는 것

*Cost function
 : 미분이 항상 가능해야함 (역전파를 위함)
 : 안될 경우 강화 학습 사용

**Comment by YunjeongPARK1** [comment]
CNN의 기초

1. CNN : Convolutional Neural Network
 	*Convolution : 주름, 복잡한 구조

2. Convolution + Subsampling + Full Connection(/Fully connected layer)

3. convolutions / subsampling => feature extraction |-> classifier(fully connected layer)

4. Why so powerful?
 	(1) Local Invariance : '동일한' 필터가 이미지 '전체'를 돌아다닌다. 이로 인해 물체가 어디에 있는지는 중요하지 않게 된다.
	(2) Compositionality : 여러 층을 이루어 Convolution, subsampling이 일어나는 것.

5. What is 'Convolution'?
	: 내가 가지고 있는 필터와 지금 만나고 있는 이미지 부분이 '얼마나 유사한가'를 평가하여 feature maps를 만드는 것
	*이때 필터는 가지고 있는 데이터로 미리 학습시켜서 만든다.

6. (1) zero-padding : 가장자리도 convolution할 수 있게 0을 가장자리에 덧대는 것
       (2)stride : 얼마 간격으로 띄워서 convolution을 진행할 것인가

7. Conv2D
        tf.nn.con2d(input, filter, strides, padding, use_cudnn_on_gpu=None, name = None)

		input : [batch, in_height, in_width, in_channel]
		*batch : 이미지 개수 
		*in_channel : 이미지의 차원
		filter : [filter_height, filter_width, in_channels, out_channels]
		*in_channels : input의 in_channel과 같다
		*out_channels : filter의 개수

8. convolution = convolution + bias + activation(ReLU)

9. subsampling : max pooling, average pooling etc
